models/match.py

```diff
@@ -0,0 +1,50 @@
+"""Module containing the Match class and other related components"""
+
+from __future__ import annotations
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+from dataclasses import dataclass
+
+
+@dataclass
+class Match:
+    """A Premier League match between two teams in a particular season.
+
+    Instance Attributes:
+        - teams: A set of the the two teams names that played this match.
+        - date_time: The date and time of the match played.
+        - winner: The name of the winning team of the match, or None if the match ended in a draw.
+        - details: A mapping from each team name to its corresponding match details.
+
+    Representation Invariants:
+        - len(teams) == 2
+    """
+    teams: set[str]
```

Comment:
Thinking that maybe we should make the teams a set of the `Team` class? In that case, winner, details would need to be fixed.

Reply:
Went that way initially but that caused a circular import error. It's possible only if the Team and Match class are defined in the same file. Any thoughts on what would be better?

Reply:
Circular imports can just be fixed by using `import ...` instead of `from ... import ...` according to StackOverflow.

Reply:
Alright. But any reason to use the Team class over Team name?

Reply:
The team name is slightly informal tbh if we alr have a team class ready to go.